lib/utils.js

```diff
@@ -559,47 +559,6 @@ exports.dQuote = function(str) {
  */
 exports.noop = function() {};
 
-/**
- * Creates a map-like object.
- *
- * @description
- * A "map" is an object with no prototype, for our purposes. In some cases
- * this would be more appropriate than a `Map`, especially if your environment
- * doesn't support it. Recommended for use in Mocha's public APIs.
- *
- * @public
```

Comment:
This is a public function. So theoretically we should soft deprecate it first ...
I don't think that many people are using it.

Reply:
Good point. That need to be corrected

Reply:
I don't see why this function has to be public. We should remove it from our API.